### PR TITLE
[GH-807] - Add name to map's configurations

### LINKED ISSUE #807: [FEATURE] - Configurator WebApp: Add name to map configs
If we want to add more maps to the game, it would be useful to differentiate map configurations by name too.

We will probably have to refactor configurator map's config since now we'll want to pick random amongst all maps (this could be done in Arena application)

### PR DESCRIPTION
**Pull request by tvillegas98**
> [!warning]
> Merge alongside [this client PR](https://github.com/lambdaclass/champions_of_mirra/pull/1996)

## Motivation

Differentiate map's by name and pick a map random amongst them all.

Closes #807 

## Summary of changes

- [Add map's name to messages.proto](https://github.com/lambdaclass/mirra_backend/commit/eea9d24d11aa66fa9feef22d425a58435264ebad)
- [Add map's name to map schema](https://github.com/lambdaclass/mirra_backend/commit/65cf169878aae32abf2a274797777c8c7b552dd1)
- [Refactor configurator map endpoint's logic to return all maps since the client is going to pick a random one among them](https://github.com/lambdaclass/mirra_backend/commit/d0bb346922b73c7ec9378a4e879eb2ebcaf8397d)
- [Allow to edit and add name to map configurations. Also display map's name in views](https://github.com/lambdaclass/mirra_backend/pull/808/commits/db78739b5e228a842d22f568c8890bfeaf8e9214)
## How to test it?

Two ways:

1. Doing a Curl
    - `curl -X GET  http://localhost:4001/curse/configuration/map` And you'll receive a list of maps (ba dum tss)

2. Playing
   - Add this line of code `Debug.Log(GameServerConnectionManager.Instance.config.Map);` in `client/Assets/Scripts/CustomLevelManager.cs:50`
   - Start a game a you'll see in Unity's shell the map's name 

## Checklist
- [x] Tested the changes locally.
- [ ] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by agustinesco — DISMISSED** [review]
LGTM!

File: apps/configurator/lib/configurator_web/controllers/map_configuration_html/show.html.heex
Comment (AminArria) [inline apps/configurator/lib/configurator_web/controllers/map_configuration_html/show.html.heex:1]: In the top of this file (and other map views) we are putting 

```
Map configuration <%= @map_configuration.id %>
```

Can you change it to 
```
Map configuration <%= @map_configuration.name %>
```

File: apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+defmodule GameBackend.Repo.Migrations.AddNameToTheMap do
+  use Ecto.Migration
+
+  def change do
+    alter table(:map_configurations) do
+      add :name, :string
+    end
+  end
```
Comment (AminArria) [inline apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs:8]: Would be nice to have an `execute/2` here that will set the map name to `Araban` so we don't need to remember to do it ourselves

**Review by AminArria — CHANGES_REQUESTED** [review]
Minor things, but overall looks good

File: apps/configurator/lib/configurator_web/controllers/map_configuration_html/show.html.heex
Comment (tvillegas98) [reply apps/configurator/lib/configurator_web/controllers/map_configuration_html/show.html.heex:1]: Sure think!

File: apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs (hunk 1 above)
Comment (tvillegas98) [reply apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs:8]: I think I can achieve it by updating all map configurations 😆

File: apps/configurator/lib/configurator_web/controllers/map_configuration_html/map_configuration_form.html.heex [hunk 2]
```diff
@@ -2,6 +2,7 @@
   <.error :if={@changeset.action}>
     Oops, something went wrong! Please check the errors below.
   </.error>
+  <.input field={f[:name]} type="text" label="Name" step="any" />
```
Comment (agustinesco) [inline apps/configurator/lib/configurator_web/controllers/map_configuration_html/map_configuration_form.html.heex:5]:
```suggestion
  <.input field={f[:name]} type="text" label="Name" />
```

File: apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs (hunk 1 above)
Comment (agustinesco) [reply apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs:10]: I think that if you add an execute statement in a change method you can't rollback the migration right? maybe you need to specify an `up` and  `down` method

Comment (AminArria) [reply apps/game_backend/priv/repo/migrations/20240725215038_add_name_to_the_map.exs:10]: `execute/2` lets you do the rollback (up/down), for this particular case some thing like

```elixir
execute(
  "update map_configurations set name = 'Araban'", 
  ""
)
```

PS: `execute/2` written from thin air so verify if plan to use

**Review by AminArria — APPROVED** [review]

**Review by agustinesco — APPROVED** [review]
LGTM!

**Pull request merged**